homeassistant/components/squeezebox/__init__.py

```diff
@@ -117,15 +127,53 @@ async def async_setup_entry(hass: HomeAssistant, entry: SqueezeboxConfigEntry) -
     )
     _LOGGER.debug("LMS Device %s", device)
 
-    coordinator = LMSStatusDataUpdateCoordinator(hass, lms)
+    server_coordinator = LMSStatusDataUpdateCoordinator(hass, lms)
 
     entry.runtime_data = SqueezeboxData(
-        coordinator=coordinator,
+        coordinator=server_coordinator,
         server=lms,
     )
 
-    await coordinator.async_config_entry_first_refresh()
+    # set up player discovery
+    hass.data.setdefault(DOMAIN, {})
+    known_players = hass.data[DOMAIN].setdefault(KNOWN_PLAYERS, [])
+
+    async def _player_discovery(now: datetime | None = None) -> None:
+        """Discover squeezebox players by polling server."""
+
+        async def _discovered_player(player: Player) -> None:
+            """Handle a (re)discovered player."""
+            if player.player_id in [p.player.player_id for p in known_players]:
+                await player.async_update()
+                async_dispatcher_send(
+                    hass, SIGNAL_PLAYER_REDISCOVERED, player.player_id, player.connected
+                )
```

Comment:
Why dont we update the coordinator? Ideally we never reference the created entities at all after creating

Reply:
`player.async_update()` calls the API not the HA entity. But I think you are right that this call is unnecessary because we are going to make that API call from the coordinator in its `_async_update_data` method.